$MirOS$
$NetBSD: patch-ba,v 1.1 2008/06/26 19:46:56 jmcneill Exp $

	First chunk is needed, since we either get
	│m4: unexpected end of input, unclosed parenthesis:
	│   configure.ac at line 5
	│autom4te-2.61: /usr/bin/m4 failed with exit status: 1
	or (with the change)
	│m4: syntax error in expr 61-M98P5809-1 - 60
	and a successful build.

	We also need -Z to disable W^X if messing around with
	section ordering, like GNU Emacs does

--- configure.ac.orig	Tue Jun 17 14:07:08 2008
+++ configure.ac	Wed Jul 30 00:06:43 2008
@@ -2,7 +2,7 @@ dnl Process this file with autoconf to p
 dnl Original author: Michael Patra
 dnl See ChangeLog file for detailed change history.
 
-m4_define(WINE_VERSION,regexp(m4_include(VERSION),[version \([-.0-9A-Za-z]+\)],[\1]))
+m4_define([WINE_VERSION], [1.0])
 
 AC_PREREQ(2.53b)
 AC_INIT([Wine],[WINE_VERSION],[wine-devel@winehq.org])
@@ -525,7 +525,13 @@ case $host_os in
       case $host_cpu in
         *i[[3456789]]86* | x86_64)
           WINE_TRY_CFLAGS([-Wl,--section-start,.interp=0x7bf00400],
-                          [LDEXECFLAGS="$LDEXECFLAGS -Wl,--section-start,.interp=0x7bf00400"])
+                          [LDEXECFLAGS="$LDEXECFLAGS -Wl,--section-start,.interp=0x7bf00400"
+			   case $host in
+			   *-*-mirbsd* | *-*-openbsd*)
+				LDEXECFLAGS="$LDEXECFLAGS -Z"	# against W^X
+				;;
+			   esac
+			  ])
           ;;
       esac
 
@@ -1321,6 +1327,9 @@ case $host_cpu in
       freebsd*)
         AC_SUBST(MAIN_BINARY,"wine-freebsd")
         AC_SUBST(EXTRA_BINARIES,"wine-pthread") ;;
+      netbsd*)
+        AC_SUBST(MAIN_BINARY,"wine-kthread")
+        AC_SUBST(EXTRA_BINARIES,"wine-pthread") ;;
     esac
     ;;
 esac
@@ -1437,7 +1446,11 @@ dnl **** Check for types ****
 
 AC_C_CONST
 AC_C_INLINE
-AC_CHECK_TYPES([mode_t, off_t, pid_t, size_t, ssize_t, long long, fsblkcnt_t, fsfilcnt_t])
+AC_CHECK_TYPES([mode_t, off_t, pid_t, size_t, ssize_t, long long])
+AC_CHECK_TYPES([fsblkcnt_t, fsfilcnt_t],,,[AC_INCLUDES_DEFAULT()
+#ifdef HAVE_SYS_STATVFS_H
+#include <sys/statvfs.h>
+#endif])
 AC_CHECK_TYPES([sigset_t],,,[#include <signal.h>])
 AC_CHECK_TYPES([request_sense],,,[#include <linux/cdrom.h>])
 
